hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -173,19 +173,22 @@ protected void createAndLinkTokenRels(
         // Link the new token relations to the account
         linkTokenRels(account, newTokenRels, tokenRelStore);
 
-        // Now replace the account's old head token number with the new head token number. This is
-        // how we link the new tokenRels to the account
-        final var firstOfNewTokenRels = newTokenRels.get(0);
-        final var updatedAcct = account.copyBuilder()
-                // replace the head token number with the first token number of the new tokenRels
-                .headTokenId(firstOfNewTokenRels.tokenId())
-                // and also update the account's total number of token associations
-                .numberAssociations(account.numberAssociations() + newTokenRels.size())
-                .build();
-
-        // Save the results
-        accountStore.put(updatedAcct);
-        newTokenRels.forEach(tokenRelStore::put);
+        // FUTURE - We may need to return a proper error status when tokens are empty
+        if (!newTokenRels.isEmpty()) {
```

Comment:
I think we have to include the call of `linkTokenRels()` (one line above). If `newTokenRels` is empty, the remove-call in line 220 fails, doesn't it?

Reply:
Good catch. I was using an account without previous relations, so the remove was never executed.
I will create a new PR to develop and cherry-pick here. 
Thanks